Makefile

```diff
@@ -50,12 +50,11 @@ hlint:
 	find src exec tests -name '*.hs' | xargs $(HLINT)
 
 test-everything:
-	perl -i -pe 's/ghc-options: -Wall/ghc-options: -Wall -Werror/' package.yaml
-	hpack
-	cabal install --enable-test
+	cabal install --enable-test --ghc-options="-Werror -Wall -O0"
+	cabal test
+	cabal haddock
```

Comment:
Possibly?  Different PR, though.  I also want to have explicit export lists in all modules, and section headings only in the export lists.  This line here is just to make sure the syntax of the comments does not deteriorate too much.